### PR TITLE
Improving Windows compatibility

### LINKED ISSUE #584: Blender & MORSE Python version matching requirement
At line 300 of the morse script there is a call to check if MORSE and Blender were built with the same Python version:

our_python_version = '.'.join((str(x) for x in sys.version_info[:3]))

Matching the Major & Minor & Micro numbers is a strong constraint as the python version installed on the system doesn't necessarily exactly match the version shipped with Blender.
Moreover, this doesn't check the version of Python MORSE was built with, but the version of the Python interpreter running MORSE.

I've found that cheating on the micro number to match the version installed on the system:

our_python_version = '3.4.X'

doesn't prevent MORSE and PyMorse to run nicely (OSX, Python 3.4.0, Blender Python 3.4.0, System Python 3.4.1).

Is there a reason for the exact version matching?
Could we at least relax the matching constraint on the micro number and issue warnings instead of stopping the execution?

### PR DESCRIPTION
**Pull request by nicolaje**
Improving Windows and Morse compatibility.

Now, when running Morse on Windows, this error is thrown:

```
C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install>found bundled pyt
hon: C:\Program Files\Blender Foundation\Blender\2.72\python
Error: Cannot read file 'C:\Users\jeremy_nicola\Desktop\morse-git\build\install\
install\Files\Blender': Permission denied
Error: Cannot read file 'C:\Users\jeremy_nicola\Desktop\morse-git\build\install\
install\Foundation\Blender\blender.exe': No such file or directory
read blend: C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\share
/morse/data/morse_default_autorun.blend
Error: Failed to read blend file 'C:\Users\jeremy_nicola\Desktop\morse-git\build
\install\install\share\morse\data\environments\sandbox.blend', not a blend file

Unable to create simulation scene. Check builder script for typos.
Execution terminated!
Error: Not freed memory blocks: 215
```
Any idea? The solution seems to be close.
Also fixes #584

### CODE REVIEW

**Comment by severin-lemaignan** [comment]
Thanks a lot for these improvements! Could you remove the last 3 commits (merge commits + version checking) from the pull request, though? They do not seem related to the Windows support.

Regarding the errors you still have, do the file ` C:\Users\jeremy_nicola\Desktop\morse-git\build
\install\install\share\morse\data\environments\sandbox.blend` actually exist?

**Comment by nicolaje** [comment]
I will edit the pull request when I come back on monday.
Unfortunately, the files do exist.
Also, I don't know what/why is looking for a Blender install in Files\Blender.
Adding a Blender installation there doesn't help: a permission denied is still thrown, Blender is launched, and chrashes.
Is there a debug mode for Blender or something to grasp more info?

**Comment by severin-lemaignan** [comment]
Yes, I was about to ask: does blender (alone!) start correctly? what does it displays on the command line at startup?

**Comment by nicolaje** [comment]
I just saw that three Blender Installation directories are accessed:
* C:\Program Files\Blender Foundation\Blender\2.72 the correct one
* C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\Files\Blender ??
* C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\Foundation\Blender\blender.exe ??

the second and the third seems to result from a "space" not being interpreted correctly, as in:
prefix+"Files/Blender Foundation/Blender/blender.exe"
resulting in : prefix+"Files/Blender" and prefix+"Foundation/Blender/blender.exe"

Do you know what could be looking for a Blender installation in "Files" ?

Otherwise, Blender alone starts correctly on the command line. Running:

```
C:\Users\jeremy_nicola>C:\Users\jeremy_nicola\Desktop\morse-git\build\install\in
stall\Foundation\Blender\blender.exe C:\Users\jeremy_nicola\Desktop\morse-git\bu
ild\install\install\share\morse\data\environments\sandbox.blend
```
gives:
```
found bundled python: C:\Users\jeremy_nicola\Desktop\morse-git\build\install\ins
tall\Foundation\Blender\2.72\python
Error: File format is not supported in file 'C:\Users\jeremy_nicola\Desktop\mors
e-git\build\install\install\share\morse\data\environments\sandbox.blend'
```
so there seems to be a problem with the sandbox environment.
Running:
```
C:\Users\jeremy_nicola>C:\Users\jeremy_nicola\Desktop\morse-git\build\install\in
stall\Foundation\Blender\blender.exe C:\Users\jeremy_nicola\Desktop\morse-git\bu
ild\install\install\share\morse\data\environments\outdoors.blend
```
gives:
```
found bundled python: C:\Users\jeremy_nicola\Desktop\morse-git\build\install\ins
tall\Foundation\Blender\2.72\python
read blend: C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\share
\morse\data\environments\outdoors.blend
Read library:  'C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\s
hare\morse\data\props\crates.blend', '//../props/crates.blend'
```
and the outdoor environment show up correctly...

So, in the builder script, replacing:
```
env = Environment('sandbox', fastmode = False)
```
by
```
env = Environment('outdoors', fastmode = False)
```

and, with the PYTHONPATH correctly pointing to Lib\site-package,  running
```
C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install>python bin\morser
un.py run test
```
gives:
```
C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install>found bundled pyt
hon: C:\Program Files\Blender Foundation\Blender\2.72\python
Error: Cannot read file 'C:\Users\jeremy_nicola\Desktop\morse-git\build\install\
install\Files\Blender': Permission denied
Error: File format is not supported in file 'C:\Users\jeremy_nicola\Desktop\mors
e-git\build\install\install\Foundation\Blender\blender.exe'
read blend: C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\share
/morse/data/morse_default_autorun.blend
Info: Read library:  'C:\Users\jeremy_nicola\Desktop\morse-git\build\install\ins
tall\share\morse\data\props\crates.blend', '//../props/crates.blend'
Read library:  'C:\Users\jeremy_nicola\Desktop\morse-git\build\install\install\s
hare\morse\data\props\crates.blend', '//../props/crates.blend'
[ERROR][MORSE] Uncaught exception, quit Blender.
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 13, in <module>
PermissionError: [WinError 32] Le processus ne peut pas acc├®der au fichier car
ce fichier est utilis├® par un autre processus: 'C:/Users/JEREMY~1/AppData/Local
/Temp/tmpgmh6ogla'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "<string>", line 13, in <module>
PermissionError: [WinError 32] Le processus ne peut pas acc├®der au fichier car
ce fichier est utilis├® par un autre processus: 'C:/Users/JEREMY~1/AppData/Local
/Temp/tmpgmh6ogla'
```
as if the temporary Python script was still locked by the morserun.py.

So, to sum up:
* who is looking for a Blender installation in "Files"?
* sandbox seems to contain something Blender can't deal with on Windows
* the temporary Python script is not correctly "released" by morserun.py

**Comment by adegroote** [comment]
I suspect that the first issue comes from ``retrieve_blender_from_path`` (through splitlines() seems ok for our uses). I suspect that we manipulate paths everywhere as strings without whitespace, so potentially all strings methods in blender.in are suspects).

For point 2, can you try to open manually the scene from Blender and see if it works (or which errors happens in Blender). For me, it is a Blender issue, but we need more information to diagnostic the issue.

For point 3, it is probable,  as all system except Windows is able to open a file by two different processus. What is the content of this file ? From a quick reading of morse.in, the two temporary file are properly closed before executed by another processus.

**Comment by nicolaje** [comment]
I removed the irrelevant commits.
So as of now, when blender is located in a whitespace-free path (C:\Blender for example), with any environment except "sandbox", and if the TMP python scripts are not "unlinked", blender opens, the scene shows up with Morsy, and Blender quits without any error message.
Any idea on how to get further?

**Comment by adegroote** [comment]
What does mean "TMP python scripts are not "unlinked" ?
Do you consider the different point I pointed to you previously ?
How do you run a Morse scene exactly if 'morse run xxx' doesn't work properly ? Any log output from the Morse console ?

**Comment by adegroote** [comment]
I merged maxint pull-request. Can you test the current master, and rebase your pull-request on top of it ? (It is probably better to close this one and open a new one) ?

Moreover, I opened a meta-issue #641 to track Windows specific bug and pull-request, please comment on this thread the known state of Morse on Windows.

Thanks.

**Comment by nicolaje** [comment]
Before this patch "morse run" or "morse edit" would for example return:

> C:\Users\jeremy_nicola\Desktop\morse-master\build\install\install>morse edit win-test
> * Adding <C:\Users\jeremy_nicola\Desktop\morse-master\build\install\install\win-test> to PYTHONPATH and MORSE_RESOURCE_PATH.
>
> * Using default scene in environment <win-test>.

> * morse UNKNOWN-dirty

> * MORSE has only been tested on Linux. It may work on other operating systems as well, but without any guarantee
>
> * Blender emitted errors during launch:
> AL lib: (EE) UpdateDeviceParams: Failed to set 44100hz, got 48000hz instead
>
>
> usage: morserun.py [-h] [-c] [--reverse-color] [-v]
>                  {create,import,rm,add,check,run,edit} ...

and then nothing would happen,
after this patch "morse edit" correctly launches blender, then if "p" is pressed the simulation is launched correctly, while "morse run" lauches blender, but as soon as the simulation starts, blender exits.

**Comment by adegroote** [comment]
I tried to rework your commits, as your rebase reverts some changes previously made. Can you check 
if https://github.com/adegroote/morse/tree/for_windows2 works for you ?

Thanks.

**Comment by nicolaje** [comment]
I confirm your for_windows2 builds and behaves as expected (morse run quits, morse edit then "p" works)

**Comment by adegroote** [comment]
I merged my for_windows2 branch, can I close the pr ?

**Comment by nicolaje** [comment]
Of course :)
